README.md

```diff
@@ -207,31 +162,125 @@ Users want to determine several types of the formatting width as following
   + fr-FR
     - 1 heure, 46 minutes et 40 secondes
 
-### Design
+#### Design
+
+Adding the locale in string format as a first argument, or specifying a ranked list of locales as an array of string locale values.
+
+### Display fractional values
+
+Sometimes it is desirable to display the smallest sub-second unit not by itself but as a fraction of the immediately larger unit.
 
-Adding the locale in `string` format as a first argument.
+#### Design
 
-# API design
+We allow users to specify a `fractionalDigits` option that will display the smallest sub-second unit with display set to `"auto"` as a fraction of the previous unit if it is non-zero. The number of digits used will be the value passed to this option.
+
+#### Example
+
+```javascript
+new Intl.DurationFormat('en', { fractionalDigits: 2 }).format('PT12.3456S'); // => 12.34 sec
+new Intl.DurationFormat('en', { milliseconds: 'narrow', fractionalDigits: 2 }).format('PT12.3456S'); // => 12 sec 345.60 ms
+```
+
+## API design
 
 ``` javascript
-    const formatter =
-        new Intl.DurationFormat('en-US', {
-            fields: [
-                'hour',
-                'minute',
-                'second'
-            ],
-            style: 'short',
-            hideZeroValues: 'none',
-        });
-
-    const duration =
-        Temporal.Duration.from({
-            hours: 2,
-            minutes: 46,
-            seconds: 40
-        });
-
-    console.log(formatter.format(duration));
-    // output: 2 hr 46 min 40 sec
+    new Intl.DurationFormat('en').format(Temporal.Duration.from('PT2H46M40S')); // => 2 hr 46 min 40 sec
+    new Intl.DurationFormat('en', {
+      hours: 'numeric',
+      seconds: 'numeric',
+    }).format(Temporal.Duration.from('PT2H40S')); // => 2:00:40
+```
+
+### Constructor
+
+#### Syntax
+
+```javascript
+new Intl.DurationFormat(locales, options)
 ```
+
+#### Parameters
+
+* `locales` (`Array` | `String`): A locale string or a list of locale strings in decreasing order of preference.
+* `options` (optional `Object`): An object for configuring the behavior of the instance. It may have some or all of the following properties:
```

Comment:
```suggestion
* `options` (optional `object`): An object for configuring the behavior of the instance. It may have some or all of the following properties:
```